blocks/rich-text/index.js

```diff
@@ -502,10 +473,12 @@ export class RichText extends Component {
 				const index = dom.nodeIndex( selectedNode );
 				const beforeNodes = childNodes.slice( 0, index );
 				const afterNodes = childNodes.slice( index + 1 );
-				const beforeElement = nodeListToReact( beforeNodes, createTinyMCEElement );
-				const afterElement = nodeListToReact( afterNodes, createTinyMCEElement );
 
-				this.restoreContentAndSplit( beforeElement, afterElement );
+				const { format } = this.props;
+				const before = format === 'string' ? domToString( beforeNodes ) : domToElement( beforeNodes );
```

Comment:
There's a lot of ternaries in this file now and it seems like it'd be easy to make mistakes in future maintenance.

I wonder if a simple improvement might just be to encapsulate logic to a single function:

```js
domToFormat( nodes, format )
```